engine/api/application_variable.go

```diff
@@ -103,10 +103,11 @@ func restoreAuditHandler(w http.ResponseWriter, r *http.Request, db *gorp.DbMap,
 		return err
 	}
 
-	if err := sanity.CheckApplication(tx, p, app); err != nil {
-		log.Warning("restoreAuditHandler: Cannot check application sanity: %s\n", err)
-		return err
-	}
+	go func() {
+		if err := sanity.CheckApplication(tx, p, app); err != nil {
```

Comment:
don't put a transaction in a goroutine; you have to start a new one after commit the previous one.
you don't want your goroutine to rollback the whole tx if timeout excedeed